### ISSUE TITLE
Coloring is not restored when using PromptText and a custom prompt

### ISSUE THREAD

**Issue by JustinGrote** [PowerShell/PSReadLine#986]
Environment data
----------------
PS version: 6.2.2                                                                                                       
PSReadline version: 2.0.0-beta4                                                                                         
os: 10.0.18941.1001 (WinBuild.160101.0800)                                                                              PS file version: 6.2.2.0                                                                                                
BufferWidth: 120                                                                                                       
BufferHeight: 30     

Steps to reproduce or exception report
--------------------------------------

PSReadline uses the Error color with -PromptText to change color to show a syntax error, however when this reverts, it seems to reset to default colors rather than restore the prompt as it was previously.

![PSReadLineIssue](https://user-images.githubusercontent.com/15258962/62253096-a6c0aa00-b3a9-11e9-91dc-d57caa90ce96.gif)

Expected Behavior
--------------------------------------
**Ideal**: PSReadline remembers what ANSI codes were used for the characters were that it replaced and respects them. This would probably be a regex of the prompt string to find the last relevant ANSI code combinations.

**Less Ideal**: PSReadline doesn't seem to remember ANSI codes if they are placed in PromptText. If it did, then the coloring expected for the prompt could be defined there and "replayed" when PSReadline reverts from the error coloring.

**Least Ideal**: PSReadline reverts the colors to a defined option such as DefaultTokenColor

**Comment by lzybkr** [comment]
It's not possible for PSReadLine to remember the colors used when originally writing the prompt. The prompt function can write to the console directly and PSReadLine can't scrape the screen to see what was written.

Maybe the right fix isn't to think of changing the prompt color, but instead specifying a error prompt text in the same way that you might specify the prompt text:

```PowerShell
Set-PSReadlineOption -PromptText "`e[44m>`e[0m " -ErrorPromptText "`e[40m>`e[0m "
```
Regardless, I think PSReadLine would need to recognize ANSI escape sequences in the prompt text so it can properly move the cursor back the right number of characters, in this example, 2.

**Comment by JustinGrote** [comment]
@lzybkr I'd be OK with this, I didn't realize prompttext just goes back x amount of spaces and rewrites rather than looking for that specific prompttext.

**Comment by JustinGrote** [comment]
https://github.com/PowerShell/PSReadLine/blob/3ed06e11dc6da4719ac69a756122bd482dc00fe0/PSReadLine/Render.cs#L380

So I assume this is what needs to be updated to recognize ANSI escapes? I'm not a C# guy but I could take a stab at a PR if this isn't on your radar anytime soon.

**Comment by lzybkr** [comment]
Yep. PowerShell has the necessary code, but not as a public API. See this [open PR](https://github.com/PowerShell/PowerShell/pull/10208) updating that exact code, you could probably just copy it more or less as is.

**Comment by JustinGrote** [comment]
I guess flag this as an upstream issue then :)

**Comment by lzybkr** [comment]
The function I pointed to above already gracefully ignores VT sequences.  I just pushed a branch (fix_986) that I think fixes this issue with no new options. Instead, you can pass 2 strings to `-PromptText`, the first being the normal prompt and the second the error, and if either of those strings has escape sequences, the configured error color will be ignored.

I think this is essentially not a breaking change despite changing the parameter type - PowerShell handles the conversion from a single string to an array w/o issue.

After I do more testing, I'll update the docs and open a PR.

**Comment by JustinGrote** [comment]
@lzybkr  thanks for looking into it! I will test when I have time. This may also resolve an occasional issue I have where a custom PromptText causes weird initial prompt issues (like it copying a  "c" from a command I was typing or whatnot, next time it happens I'll capture it)

**Comment by daxian-dbw** [comment]
Closed via #1180
